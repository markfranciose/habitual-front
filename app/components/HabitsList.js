import React, { Component } from 'react';
import { FlatList, Text, View } from 'react-native';
import styles from './ComponentStyles';

export default class HabitsList extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Your Habits</Text>
        <FlatList
          data={this.props.habits}

          renderItem={({item}) => <Text style={styles.habit}>{item.name}</Text>}
        />
      </View>
    );
  }
}
