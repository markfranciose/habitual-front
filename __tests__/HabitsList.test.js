import 'react-native';
import React from 'react';
import HabitsList from '../app/components/HabitsList';
import renderer from 'react-test-renderer';

describe('habits list', () => {

  test("is true", () => {
    expect(true).toBe(true);
  });

  test('renders the habits list correctly', () => {
    const tree = renderer.create(<HabitsList />).toJSON();
      expect(tree).toMatchSnapshot();
  });

});
