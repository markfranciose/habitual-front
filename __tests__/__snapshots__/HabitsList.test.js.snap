// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habits list renders the habits list correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 19,
        "fontWeight": "bold",
        "marginTop": 22,
        "textAlign": "center",
      }
    }
  >
    Your Habits
  </Text>
  <RCTScrollView
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
